### PR TITLE
fix: read list of pnpm patches from lock file when update_pnpm_lock is False

### PR DESCRIPTION
**Pull request by gregmagolan**
When `update_pnpm_lock` is `False` then we can read the list of packages from the lock file since that won't break the update pnpm lock workflow when it is not enabled. This means that there is no phantom read of an undeclared package.json file for users in this case.

Fixes https://github.com/aspect-build/rules_js/issues/910.

Also related to Bazel Slack conversation: https://bazelbuild.slack.com/archives/CEZUUKQ6P/p1682614797831909

### Type of change

- Bug fix (change which fixes an issue)

### Test plan

- Covered by existing test cases

### CODE REVIEW

**Review by kormide — APPROVED** [review]
🆒

File: npm/private/npm_translate_lock_state.bzl [hunk 1]
```diff
@@ -56,11 +59,11 @@ WARNING: `update_pnpm_lock` attribute in `npm_translate_lock(name = "{rctx_name}
     if pnpm_lock_exists:
         _load_lockfile(priv, rctx, label_store)
 
-    if not root_package_json_declared:
-        # If a root package.json label wasn't declared then we didn't load pnpm.patchedDependencies
-        # above. Try again now that the lockfile has been read so that we can derive the root
-        # package.json from the `importers` field.
-        _init_patched_dependencies_labels(priv, rctx, label_store)
+    # If we didn't initialize the patched dependency labels above then try again now. We'll either
```
Comment (jbedard) [inline npm/private/npm_translate_lock_state.bzl:62]: "if we didn't ... then try again" - don't we try again every time now?

**Review by jbedard — APPROVED** [review]

File: npm/private/npm_translate_lock_state.bzl (hunk 1 above)
Comment (gregmagolan) [reply npm/private/npm_translate_lock_state.bzl:62]: I simplified the logic so the function itself marks itself as called already and exits early

Comment (jbedard) [reply npm/private/npm_translate_lock_state.bzl:62]: Should the comment change then?

Comment (gregmagolan) [reply npm/private/npm_translate_lock_state.bzl:62]: Sure. I'll re-phrase so it is more obvious

Comment (gregmagolan) [reply npm/private/npm_translate_lock_state.bzl:62]: Actually, I'll move the "already intialized" state to this function so it reads clearly

Comment (jbedard) [reply npm/private/npm_translate_lock_state.bzl]: > Actually, I'll move the "already intialized" state to this function so it reads clearly

sgtm 👍

**Pull request merged**